tools/generate-user-messages-screenshot

```diff
@@ -0,0 +1,197 @@
+        internal_prep_stream_message(
+            subscribers[message["sender"]],
+            stream,
+            topic,
+            message["content"],
+        )
+        for message in staged_messages
+    ]
+
+    message_ids = [
+        sent_message_result.message_id for sent_message_result in do_send_messages(messages)
+    ]
+    global message_thread_ids
+    message_thread_ids += message_ids
+
+    for message, message_id in zip(staged_messages, message_ids):
+        if message.get("reactions") is not None:
+            reactions = message["reactions"]
+            for reaction, user_names in reactions.items():
+                users = [subscribers[user_name] for user_name in user_names]
+                add_message_reactions(message_id, reaction, users)
+
+    return message_ids
+
+
+def add_message_reactions(message_id: int, emoji: str, users: List[UserProfile]) -> None:
+    preview_message = access_message(user_profile=DEFAULT_USER, message_id=message_id)
+    emoji_data = get_emoji_data(realm.id, emoji)
+    for user in users:
+        do_add_reaction(
+            user, preview_message, emoji, emoji_data.emoji_code, emoji_data.reaction_type
+        )
+
+
+def capture_streams_narrow_screenshot(
+    image_path: str, stream_name: str, topic: str, unread_msg_id: int
+) -> None:
+    stream = ensure_stream(realm, stream_name, acting_user=DEFAULT_USER)
+
+    narrow_uri = topic_narrow_url(realm=realm, stream=stream, topic_name=topic)
+    narrow = f"{stream_name}/{topic}"
+    screenshot_script = os.path.join(TOOLS_DIR, "thread-screenshot.js")
+    subprocess.check_call(
+        ["node", screenshot_script, narrow_uri, narrow, str(unread_msg_id), image_path, realm.url]
+    )
+
+
+parser = argparse.ArgumentParser()
+group = parser.add_mutually_exclusive_group(required=True)
+
+group.add_argument(
+    "--thread",
+    nargs="+",
+    type=str,
+    help="Path of the file where the thread for screenshot is present",
+)
+fixture_group = parser.add_argument_group("thread")
+
+options = parser.parse_args()
+prepare_puppeteer_run()
+
+realm = get_realm("zulip")
+with open(options.thread[0]) as f:
+    threads = json.load(f)
+    for thread in threads:
+        for user, avatar in thread["users"].items():
+            create_user(user, avatar)
+
+        if thread["recipient_type"] == "channel":
+            users = list(thread["users"].keys())
+            users.append("Iago")
+            create_and_subscribe_stream(thread["channel"], users, thread["color"])
+            message_ids = send_stream_messages(
+                thread["channel"], thread["topic"], thread["messages"]
+            )
+            capture_streams_narrow_screenshot(
+                thread["screenshot"], thread["channel"], thread["topic"], min(message_ids)
+            )
+
+
+thread_messages = [
+    access_message(user_profile=DEFAULT_USER, message_id=message_id)
+    for message_id in message_thread_ids
+]
+do_delete_messages(realm, thread_messages)
```

Comment:
I am deleting all the messages added after the process ends because next time when you re run the script with the same data, then the previous messages would still be visible.

Reply:
Yeah, that's pretty reasonable.

Reply:
done!